internal/topology/variables/cluster_variable_validation.go

```diff
@@ -98,7 +98,7 @@ func validateClusterVariablesDefined(clusterVariables []clusterv1.ClusterVariabl
 	for i, clusterVariable := range clusterVariables {
 		if _, ok := clusterClassVariables[clusterVariable.Name]; !ok {
 			return field.ErrorList{field.Invalid(fldPath.Index(i).Child("name"), clusterVariable.Name,
-				fmt.Sprintf("variable with name %q is not defined in the ClusterClass", clusterVariable.Name))} // TODO: consider if to add ClusterClass name
+				fmt.Sprintf("variable with name %q is not reconciled in ClusterClass `status.Variables`", clusterVariable.Name))} // TODO: consider if to add ClusterClass name
```

Comment:
To be honest I prefered the previous wording. What does it mean to a user that the variable is not reconciled?

What about
```suggestion
				fmt.Sprintf("variable with name %q is not defined in ClusterClass `status.variables`", clusterVariable.Name))} // TODO: consider if to add ClusterClass name
```

Reply:
I would assume that it's the responsibility of whoever calls the func to ensure the ClusterClass has been reconciled already